ports/servoshell/desktop/cli.rs

```diff
@@ -8,13 +8,39 @@ use getopts::Options;
 use log::error;
 use servo::config::opts::{self, ArgumentParsingResult};
 use servo::servo_config::pref;
+#[cfg(feature = "tracing")]
+use tracing_perfetto::PerfettoLayer;
+#[cfg(feature = "tracing")]
+use tracing_subscriber::layer::SubscriberExt;
+#[cfg(feature = "tracing")]
+use tracing_subscriber::util::SubscriberInitExt;
+#[cfg(feature = "tracing")]
+use tracing_subscriber::EnvFilter;
 
 use crate::desktop::app::App;
 use crate::panic_hook;
 
 pub fn main() {
     crate::crash_handler::install();
 
+    #[cfg(feature = "tracing")]
+    {
+        let file = std::fs::File::create("servo.pftrace").unwrap();
+        let perfetto_layer = PerfettoLayer::new(std::sync::Mutex::new(file));
+        // Set up a custom tracing subscriber that behaves like the default fmt
+        // subscriber, but also lets us insert our own layers for things like
+        // compositor-to-constellation and script-to-constellation logging.
+        // <https://tokio.rs/tokio/topics/tracing>
+        // <https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/fmt/index.html#composing-layers>
+        let env_filter_layer = EnvFilter::from_default_env();
+        let fmt_layer = tracing_subscriber::fmt::layer();
```

Comment:
i think these can be removed if we’re just using tracing for perfetto right now?